lib/Jsrt/JsrtSourceHolder.cpp

```diff
@@ -118,20 +235,61 @@ namespace Js
         Assert(scriptLoadCallback != nullptr);
         Assert(this->mappedSource == nullptr);
 
-        const typename JsrtSourceHolderPolicy<TLoadCallback>::TLoadCharType *source = nullptr;
+        JsValueRef scriptVal;
+        JsParseScriptAttributes attributes = JsParseScriptAttributeNone;
         size_t sourceLength = 0;
         utf8char_t *utf8Source = nullptr;
         size_t utf8Length = 0;
         size_t utf8AllocLength = 0;
 
-        if (!scriptLoadCallback(sourceContext, &source))
+        ThreadContext *threadContext = ThreadContext::GetContextForCurrentThread();
+        bool isActive = threadContext->IsScriptActive();
+        // check if we are still in script land
+        if (isActive)
+        {
+            threadContext->SetIsScriptActive(false);
+            threadContext->GetRecycler()->SetIsScriptActive(false);
+        }
+
+        if (!scriptLoadCallback(sourceContext, &scriptVal, &attributes))
```

Comment:
Should scriptLoadCallback be guarded by normal BEGIN_LEAVE_SCRIPT macro because scriptLoadCallback is external function and it can call any other JsRT api before returning